testutil/testutil.go

```diff
@@ -76,6 +78,23 @@ func Equals(tb testing.TB, exp, act interface{}, v ...interface{}) {
 	tb.Fatal(sprintfWithLimit("\033[31m%s:%d:"+msg+"\n\n\texp: %#v\n\n\tgot: %#v%s\033[39m\n\n", filepath.Base(file), line, exp, act, diff(exp, act)))
 }
 
+// EqualsWithNaN fails the test if exp is not equal to act and ensures NaN == NaN.
+// Keep in mind that this implementation cannot check unexported fields in structs on its own
+// and needs setting custom options for it.
+func EqualsWithNaN(tb testing.TB, exp, act interface{}, opts cmp.Options, v ...interface{}) {
```

Comment:
I think this functionality is a bit too narrow.

I wished we could do `Equals(<things>, ...option)`. Perhaps we could do `WithCmpOpts(opts).Equal` ? and there define which comparison to using (reflect or go-cmp) and perhaps go cmp opts?  

Alternatively `EqualsWithOpts` - but I think I like that former better. Or even `With(opts).Equal` is fine too.

Reply:
Ack! Will try to modify

Reply:
So I updated this to now be `testutil.WithCmpOpts(opts).Equals` for `go-cmp` comparison, and `tesutil.Equals` for `reflect.DeepEqual` one. I wanted to preserve `testutil.Equals` as it's super clean and used everywhere right now, whereas the go-cmp one would only be needed in a few specific cases. WDYT? 🙂 

If this looks good, I might add some comment explaining this then.